packages/dcos-launch/extra/launch/config.py

```diff
@@ -323,6 +330,19 @@ def calculate_ssh_private_key(ssh_private_key_filename):
     return load_string(ssh_private_key_filename)
 
 
+def calculate_instance_ami(os_name, aws_region):
+    return test_util.aws.OS_AMIS[os_name][aws_region]
+
+
+def validate_key_name(provider, platform, key_helper):
+    if provider == 'aws' and platform == 'onprem' and key_helper == 'false':
```

Comment:
See above; we do not want the key_name validator function to reporting when platform and provider are incompatible as it will swamp the output of the validation with meaningless and misleading errors